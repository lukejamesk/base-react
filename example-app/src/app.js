import React from 'react'
import PropTypes from 'prop-types'
import {
  Typography,
  Button
} from '@lukejamesk-ui'

const App = ({ name }) => (
  <div>
    <Typography type="heading1">{name}</Typography>
    <Typography>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Doloribus perferendis molestiae.
    </Typography>
    <Button>
      My Button
    </Button>
  </div>
)

App.propTypes = {
  name: PropTypes.string.isRequired
}

export default App
