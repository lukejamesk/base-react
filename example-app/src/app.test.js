import React from 'react'
import { mount } from 'enzyme'
import {
  ThemeProvider,
  defaultTheme
} from '@lukejamesk-ui/styles'
import App from './app'

describe('<App />', () => {
  it('should render', () => {
    const wrapper = mount(
      <ThemeProvider theme={defaultTheme}>
        <App name="Test" />
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
