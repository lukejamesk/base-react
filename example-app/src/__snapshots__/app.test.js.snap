// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render 1`] = `
<App
  name="Test"
>
  <div>
    <ForwardRef
      type="heading1"
    >
      <h1>
        Test
      </h1>
    </ForwardRef>
    <ForwardRef>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Doloribus perferendis molestiae.
      </p>
    </ForwardRef>
    <ForwardRef>
      <button
        type="button"
      >
        My Button
      </button>
    </ForwardRef>
  </div>
</App>
`;
